Core/Util/GameManager.cpp

```diff
@@ -258,32 +260,37 @@ ZipFileContents DetectZipFileContents(struct zip *z, ZipFileInfo *info) {
 	}
 }
 
-bool GameManager::InstallGame(const std::string &url, const std::string &fileName, bool deleteAfter) {
+// Parameters need to be by value, since this is a thread func.
+bool GameManager::InstallGame(Path url, Path fileName, bool deleteAfter) {
 	if (installInProgress_) {
 		ERROR_LOG(HLE, "Cannot have two installs in progress at the same time");
 		return false;
 	}
 
-	if (!File::Exists(fileName)) {
+	if (!File::Exists(Path(fileName))) {
 		ERROR_LOG(HLE, "Game file '%s' doesn't exist", fileName.c_str());
 		return false;
 	}
 
+	std::string extension = url.GetFileExtension();
 	// Examine the URL to guess out what we're installing.
-	if (endsWithNoCase(url, ".cso") || endsWithNoCase(url, ".iso")) {
+	if (extension == "cso" || extension == "iso") {
 		// It's a raw ISO or CSO file. We just copy it to the destination.
-		std::string shortFilename = GetFilenameFromPath(url);
+		std::string shortFilename = Path(url).GetFilename();
 		return InstallRawISO(fileName, shortFilename, deleteAfter);
 	}
 
 	auto sy = GetI18NCategory("System");
 	installInProgress_ = true;
 
-	std::string pspGame = GetSysDirectory(DIRECTORY_GAME);
-	std::string dest = pspGame;
+	Path pspGame = GetSysDirectory(DIRECTORY_GAME);
+	Path dest = pspGame;
 	int error = 0;
+
+	// TODO(scoped): zip_open ain't gonna work. Might have to hack in one that can do it
```

Comment:
Latest libzip seems to have a `zip_fdopen()`, so maybe we just need to update.

https://libzip.org/documentation/zip_fdopen.html

-[Unknown]

Reply:
Good catch!